Update (new version) of MEF3 format (MED).

Hello,

I am Matt Stead, the creator of all the MEF formats (1-3).  The new version MEF version (MED) is faster & produces better compression than its predecessors. It continues to be HIPAA & IRB compliant. A fuller description of MED is available on https://MEDFormat.org.

I currently have 4 groups asking me about official BIDS compatibility. 

MED data can be read in open source multiple tools (Matlab, FieldTrip, EEGLab, Python , Neo).  

These tools are freely available in compiled form for all major platforms (MacOS, Linux, Windows) as well as having source code and specification documents available on https://github.com/MEDFormat.  

There are also multiple free data converters (MEF, EDF, DAT, Neuralynx (NCS & NRD), & NATUS) available.

I would be happy to make the pull request for this.

@bids-standard/raw-electrophys-ieeg

Hi Chris,

I did the changes & pull requests that you recommended, but on one it failed a check:
https://github.com/bids-standard/bids-specification/pull/1957￼
Update ieeg to include MED.yaml by MEDFormat · Pull Request #1957 · bids-standard/bids-specification
github.com

The check error is in "schemacode_ci / Validate schema”

Which gives this for the test details:
=========================== short test summary info ============================
FAILED tools/schemacode/bidsschematools/data/tests/test_rules.py::test_rule_objects - AssertionError: Undefined objects found in rules: files.deriv.preprocessed_data.ieeg_ieeg_common.extensions[0] == .medd/
files.raw.ieeg.ieeg.extensions[0] == .medd/
============ 1 failed, 1 passed, 66 deselected, 2 warnings in 7.59s ============


I looked around for that file, but I could not find it.

Any chance you can point me to it?


Cheers, Matt

￼

Matt Stead
***@***.***

> On Sep 26, 2024, at 2:05 PM, Chris Markiewicz ***@***.***> wrote:
> 
> 
> FWIW I'm +1 on this, but I don't work with these data.
> 
> I believe the places to update in the schema would be:
> 
> https://github.com/bids-standard/bids-specification/blob/master/src/schema/objects/extensions.yaml
> https://github.com/bids-standard/bids-specification/blob/master/src/schema/rules/files/raw/ieeg.yaml
> I did open MEDFormat/MEDPython#6 <https://github.com/MEDFormat/MEDPython/issues/6>, though. The usability of a GPL'd Python library is questionable if users do not want to GPL their own code. I am having trouble digging up references, but I recall seeing some lack of clarity whether Python's import mechanism triggered LGPL or not. IIRC the most similar license that was definitely safe for use in non-GPL Python projects was the Mozilla Public License v2.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/bids-standard/bids-specification/issues/1931#issuecomment-2377833243>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ATNKTHVXMNZAGHNNILQ7RI3ZYRSIPAVCNFSM6AAAAABO5W2BXKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGNZXHAZTGMRUGM>.
> You are receiving this because you authored the thread.
>

--- a/src/modality-specific-files/intracranial-electroencephalography.md
+++ b/src/modality-specific-files/intracranial-electroencephalography.md
@@ -38,6 +38,7 @@ stored in one of the following formats:
 | [EEGLAB](https://sccn.ucsd.edu/eeglab/index.php)                          | `.set`, `.fdt`           | The format used by the MATLAB toolbox [EEGLAB](https://sccn.ucsd.edu/eeglab/index.php). Each recording consists of a `.set` file with an OPTIONAL `.fdt` file.             |
 | [Neurodata Without Borders](https://nwb-schema.readthedocs.io/en/latest/) | `.nwb`                   | Each recording consists of a single `.nwb` file.                                                                                                                           |
 | [MEF3](https://osf.io/e3sf9/)                                             | `.mefd`                  | Each recording consists of a `.mefd` directory.                                                                                                                            |
+| [MED](https://medformat.org/)                                             | `.medd`                  | Each recording consists of a `.medd` directory.                                                                                                                            |
 
 It is RECOMMENDED to use the European data format, or the BrainVision data
 format. It is furthermore discouraged to use the other accepted formats over

--- a/src/schema/objects/extensions.yaml
+++ b/src/schema/objects/extensions.yaml
@@ -153,6 +153,14 @@ md:
   display_name: Markdown
   description: |
     A Markdown file.
+medd:
+  value: .medd/
+  display_name: Multiscale Electrophysiology Data Format
+  description: |
+    A directory in the [MED](https://medformat.org/) format.
+    Successor to the MEF 3.0 format.
+
+    Each recording consists of a `.medd` directory.
 mefd:
   value: .mefd/
   display_name: Multiscale Electrophysiology File Format Version 3.0

--- a/src/schema/rules/files/raw/ieeg.yaml
+++ b/src/schema/rules/files/raw/ieeg.yaml
@@ -3,6 +3,7 @@ ieeg:
   suffixes:
     - ieeg
   extensions:
+    - .medd/
     - .mefd/
     - .json
     - .edf

